CMakeLists.txt

```diff
@@ -64,8 +64,8 @@ set(BM_INCLUDES
 include_directories(${BM_INCLUDES})
 
 if(ENABLE_TESTING)
-    set(CMAKE_CXX_FLAGS "-g -Wall")
-    set(CMAKE_C_FLAGS "-g -Wall -W")
+    set(CMAKE_CXX_FLAGS "-g -Wall --coverage")
+    set(CMAKE_C_FLAGS "-g -Wall -W  --coverage")
```

Comment:
with this change it is generating a report again. I'm not sure if this is the best way to do it.
<img width="1373" alt="Screenshot 2024-11-04 at 12 12 48 PM" src="https://github.com/user-attachments/assets/99d091c3-c0e9-448e-b567-83bed72df9f9">

Reply:
Hmmm but now the windows build is failing 🤔

Reply:
oh but the last one succeeded???

Reply:
The justification for moving the `--coverage` to  here is because the drivers/adin2111 unit tests don't use the `create_gtest` function, therefore even when trying to filter out for the /test/ directory, it didn't include the adin2111 tests in the report

Reply:
cc @matt001k @towynlin I added this change to generate the report. Not sure if this is the best way to do it, but this is what I found that works.

Reply:
@victorsowa12 the intention of doing it in the function was to generate coverage individually for each test. That is each source file we are running the unit test on, we will only generate the coverage report for that source file/unit test combination.

The change you made way will create a report but will also include header files, as well as report coverage whenever a line is invoked, whether it was apart of the test specifically for that source file or not (example: if we compile the ll.c logic into a unit test for packet.c, this would also report the coverage for ll.c and not just packet.c when we only really want to report the coverage for packet.c since that is the file we are testing)

Reply:
Ahh I see! I'll try this modified method again and convert the adin2111 test to using the create_gtest function.

```
# Generate artifacts for coverage report using lcov
    file(GLOB_RECURSE FILES "../*")
    foreach(FILE ${FILES})
        if (NOT FILE MATCHES "^test/")
            get_filename_component(FILENAME "${FILE}" NAME)
            if (FILENAME MATCHES ".*${test_name}.*")
                set_source_files_properties(${FILE} PROPERTIES
                    COMPILE_FLAGS
                    "-fprofile-arcs -ftest-coverage"
                )
            endif()
        endif()
    endforeach()
```

Reply:
Hmmm okay, so I tried to use the above ^ in `create_gtest` to filter out `test/` directory files and got this:
<img width="1291" alt="Screenshot 2024-11-04 at 2 03 50 PM" src="https://github.com/user-attachments/assets/aac967b2-9b86-4ecd-8cf9-783871dc2296">

It still has some .h files. Is that expected?

Also I reverted changes I made to the adin2111 unit tests since Zac is still working on them and I don't want to accidentally change his work. Just to note tho, the adin tests are not in the coverage report as of now.